src/yio-model/mediaplayer/tvchannelmodel_mediaplayer.h

```diff
@@ -0,0 +1,134 @@
+
+#pragma once
+
+#include <QAbstractListModel>
+#include <QObject>
+#include <QVariant>
+#include <QtDebug>
+
+class TvChannelModelItem {
+ public:
+    TvChannelModelItem(const QString& key, const QString& time, const QString& title,
+                       const QString& subtitle, const QString& type,
+              const QString& imageUrl, const QVariant& commands)
+        : m_key(key), m_time(time), m_title(title), m_subtitle(subtitle),
+          m_type(type), m_imageUrl(imageUrl), m_commands(commands) {}
+
+    QString  itemKey() const { return m_key; }
+    QString  itemTime() const { return m_time; }
+    QString  itemTitle() const { return m_title; }
+    QString  itemSubtitle() const { return m_subtitle; }
+    QString  itemType() const { return m_type; }
+    QString  itemImageUrl() const { return m_imageUrl; }
+    QVariant itemCommands() const { return m_commands; }
+
+ private:
+    QString  m_key;
+    QString  m_time;
+    QString  m_title;
+    QString  m_subtitle;
+    QString  m_type;
+    QString  m_imageUrl;
+    QVariant m_commands;
+};
+
+class ListTvChannelModel : public QAbstractListModel {
+    Q_OBJECT
+    Q_PROPERTY(int count READ count WRITE setCount NOTIFY countChanged)
+
+ public:
+    enum SearchRoles { KeyRole = Qt::UserRole + 1, TimeRole, TitleRole, SubTitleRole,
+                       TypeRole, ImageUrlRole, CommandsRole };
+
+    explicit ListTvChannelModel(QObject* parent = nullptr);
+    ~ListTvChannelModel() {}
+
+    int                    count() const;
+    int                    rowCount(const QModelIndex& p = QModelIndex()) const;
+    QVariant               data(const QModelIndex& index, int role = Qt::DisplayRole) const;
+    QHash<int, QByteArray> roleNames() const;
+
+    void append(const TvChannelModelItem& o);
+
+ public slots:  // NOLINT open issue: https://github.com/cpplint/cpplint/pull/99
+    void setCount(int count);
+
+ signals:
+    void countChanged(int count);
+
+ private:
+    int              m_count;
+    QList<TvChannelModelItem> m_data;
+};
+
+class BrowseTvChannelModel : public QObject {
+    Q_OBJECT
+    Q_PROPERTY(QString id READ id NOTIFY idChanged)
+    Q_PROPERTY(QString title READ title NOTIFY titleChanged)
+    Q_PROPERTY(QString subtitle READ subtitle NOTIFY subtitleChanged)
+    Q_PROPERTY(QString type READ type NOTIFY typeChanged)
+    Q_PROPERTY(QString imageUrl READ imageUrl NOTIFY imageUrlChanged)
+    Q_PROPERTY(QObject* model READ model NOTIFY modelChanged)
+    Q_PROPERTY(QStringList commands READ commands NOTIFY commandsChanged)
+
+ public:
+    BrowseTvChannelModel(const QString& id, const QString& time,
+                         const QString& title, const QString& subtitle, const QString& type,
+                         const QString& imageUrl, const QStringList& commands = {}, QObject* parent = nullptr)
+        : m_id(id), m_time(time), m_title(title), m_subtitle(subtitle), m_type(type),
+          m_imageUrl(imageUrl), m_commands(commands){}
```

Comment:
One space away for 👍 
cpplint error: `src/yio-model/mediaplayer/tvchannelmodel_mediaplayer.h:100:  Missing space before {  [whitespace/braces] [5]`
https://github.com/YIO-Remote/integrations.library/pull/13/checks#step:5:37

Reply:
Solved